lib/init.js

```diff
@@ -0,0 +1,153 @@
+'use babel';
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { CompositeDisposable } from 'atom';
+
+let helpers;
+let flexPmd;
+let os;
+let readFileAsync;
+let unlinkFileAsync;
+
+const REGEX = /<violation beginline="(-?\d+)" endline="(-?\d+)" begincolumn="(-?\d+)" endcolumn="(-?\d+)".* priority="(-?\d+)">(.+?)<\/violation>/g;
+
+const showErrorNotification = (text = 'Unexpected error', description) => {
+  atom.notifications.addError(`linter-flexpmd ${text}`, { description });
+};
+
+const loadDeps = () => {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+  if (!flexPmd) {
+    flexPmd = require('flexpmd');
+  }
+  if (!os) {
+    os = require('os');
+  }
+  if (!readFileAsync || !unlinkFileAsync) {
+    const fs = require('fs');
+    const util = require('util');
+    readFileAsync = util.promisify(fs.readFile);
+    unlinkFileAsync = util.promisify(fs.unlink);
+  }
+};
+
+const parseFlexPmdOutput = (output, file, editor) => {
+  const messages = [];
+  let match = REGEX.exec(output);
+  while (match !== null) {
+    const line = Number.parseInt(match[1], 10) - 1;
+    const col = Number.parseInt(match[3], 10) - 1;
+    messages.push({
+      severity: 'warning',
+      excerpt: match[6],
+      location: {
+        file,
+        position: helpers.generateRange(editor, line, col),
+      },
+    });
+    match = REGEX.exec(output);
+  }
+  return messages;
+};
+
+module.exports = {
+  activate() {
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterFlexpmdDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-flexpmd');
+      }
+      loadDeps();
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterFlexpmdDeps);
+    this.idleCallbacks.add(depsCallbackID);
+
+    this.subscriptions = new CompositeDisposable();
+    this.subscriptions.add(
+      atom.config.observe(
+        'linter-flexpmd.javaExecutablePath',
+        (value) => { this.javaExecutablePath = value; },
+      ),
+      atom.config.observe(
+        'linter-flexpmd.javaMemoryAllocation',
+        (value) => { this.javaMemoryAllocation = value; },
+      ),
+    );
+  },
+
+  deactivate() {
+    this.idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
+    this.idleCallbacks.clear();
+    this.subscriptions.dispose();
+  },
+
+  provideLinter() {
+    return {
+      name: 'flexpmd',
+      grammarScopes: ['source.actionscript.3'],
+      scope: 'file',
+      lintsOnChange: false,
+      lint: async (editor) => {
+        if (!atom.workspace.isTextEditor(editor)) {
+          return null;
+        }
+
+        const filePath = editor.getPath();
+        if (!filePath) {
+          return null;
+        }
+
+        loadDeps();
+
+        const tempDir = os.tmpdir();
+        const tempPmdFile = `${tempDir}/pmd.xml`;
```

Comment:
What happens if multiple runs are triggered at the same time and they both try to write to the same file? This is why I recommended something like [`temp`](https://www.npmjs.com/package/temp) to manage it for you, since that would give you a unique directory/file for each run.